my_pyscf/mcpdft/laspdft.py

```diff
@@ -32,30 +33,53 @@ def get_h2eff(self, mo_coeff=None):
             eri = ao2mo.full(self.mol, mo_coeff, verbose=self.verbose,
                                 max_memory=self.max_memory)
         return eri
-
-def get_mcpdft_child_class(mc, ot, **kwargs):
+        
+def get_mcpdft_child_class(mc, ot, DoLASSI=False,  **kwargs):
     mc_doc = (mc.__class__.__doc__ or 'No docstring for MC-SCF parent method')
    
     class PDFT(_LASPDFT, mc.__class__):
         __doc__= mc_doc + '\n\n' + _LASPDFT.__doc__
         _mc_class = mc.__class__
-
+       
         def get_h2eff(self, mo_coeff=None):
             if self._in_mcscf_env: return mc.__class__.get_h2eff(self, mo_coeff=mo_coeff)
             else: return _LASPDFT.get_h2eff(self, mo_coeff=mo_coeff)
 
-        make_one_casdm1s=mc.__class__.make_casdm1s
-        make_one_casdm2=mc.__class__.make_casdm2
-        
+        # This code will change once state specific RDMs are available
+        def make_one_casdm1s(self, ci=None, state=0, **kwargs):
+            if self._in_mcscf_env: return mc.__class__.make_casdm1s
+            else: 
+                if self.roots_casdm1s is None: self.roots_casdm1s=self.states_make_casdm1s(ci=ci, **kwargs)
+                return self.roots_casdm1s[state, :, :, :]
```

Comment:
This is super sketchy: the distinction between LASSCF-PDFT and LASSI-PDFT behavior hinges entirely on whether there is a cached variable `roots_casdm1s` that was precomputed. Instead of doing this, can you have the code properly branch here based on the value of `DoLASSI`?